### ISSUE TITLE
Random features and more

### ISSUE THREAD

**Issue by MaxHalford** [creme-ml/creme#201]
I stumbled on [this paper](https://people.eecs.berkeley.edu/~brecht/papers/07.rah.rec.nips.pdf) which got the test of time award at NIPS'17. From what I understand they show how using "random" features can bring the same performance as a kernelized SVM. There are reasonably good implementations [here](https://github.com/tiskw/RandomFourierFeatures/blob/master/source/PyRFF.py) and [here](https://github.com/hichamjanati/srf). scikit-learn also has a [module](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.kernel_approximation) dedicated to this. Here's a quote from their documentation:

> The advantage of using approximate explicit feature maps compared to the kernel trick, which makes use of feature maps implicitly, is that explicit mappings can be better suited for online learning and can significantly reduce the cost of learning with very large datasets. Standard kernelized SVMs do not scale well to large datasets, but using an approximate kernel map it is possible to use much more efficient linear SVMs. In particular, the combination of kernel map approximations with SGDClassifier can make non-linear learning on large datasets possible.

[This](http://fastml.com/the-secret-of-the-big-guys/) article from FastML also hints at something similar. Also see [this](https://arxiv.org/pdf/1510.06664.pdf) paper.

I think it would be nice to implement the random Fourier shown in the paper. We could also implement [other kinds of kernels](http://crsouza.com/2010/03/17/kernel-functions-for-machine-learning-applications/). API-wise this would be like the `PolynomialExtender` in the `preprocessing` module. Maybe we could put all this in the `feature_extraction` module? Possibly in a submodule of it's own?

**Comment by MaxHalford** [comment]
I just implemented `preprocessing.RBFSampler`. Before closing this issue I would like to implement [Spherical Random Features for Polynomial Kernels](https://papers.nips.cc/paper/5943-spherical-random-features-for-polynomial-kernels.pdf).
